tests/test_goodconf.py

```diff
@@ -154,3 +157,46 @@ class TestConf(GoodConf):
 
     with pytest.raises(ValidationError):
         TestConf(load=True)
+
+
+def test_set_on_init():
+    class TestConf(GoodConf):
+        a: str = Field()
+
+    val = "test"
+    c = TestConf(a=val)
+    assert c.a == val
+
+
+def test_env_prefix():
+    class TestConf(GoodConf):
+        a: bool = False
+
+        class Config:
+            env_prefix = "PREFIX_"
+
+    with env_var("PREFIX_A", "True"):
+        c = TestConf(load=True)
+
+    assert c.a
+
+
+def test_precedence(tmpdir):
+    path = tmpdir.join("myapp.json")
+    path.write(json.dumps({"init": "file", "env": "file", "file": "file"}))
+
+    class TestConf(GoodConf, default_files=[path]):
```

Comment:
Did you know that you do not need to use `class Config`? Pydantic allows you to pass those via the metaclass kwargs.

Reply:
I did not!